### PR TITLE
feat: #62 create RP tenant endpoint

### LINKED ISSUE #62: rp adapter: create rp tenant endpoint
This is a part of trustbloc/edge-sandbox#382

This will allow registration of the rp-rest service.

### PR DESCRIPTION
**Pull request by llorllale**
closes #62 

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

**Comment by llorllale** [comment]
I wonder if we should extend this to also register the RP in Hydra: https://www.ory.sh/hydra/docs/reference/api/#create-an-oauth-20-client

File: pkg/restapi/rp/operation/operations.go [hunk 1]
```diff
@@ -586,3 +592,61 @@ func testResponse(w io.Writer) {
...
+	err := json.NewDecoder(r.Body).Decode(request)
+	if err != nil {
+		msg := fmt.Sprintf("failed to decode request: %s", err)
+		logger.Errorf(msg)
+		commhttp.WriteErrorResponse(w, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if request.ClientID == "" || request.Label == "" || request.PublicDID == "" {
```
Comment (rolsonquadras) [inline pkg/restapi/rp/operation/operations.go:608]: Here publicDID is part of the request.  Who calls this API ? Who creates public DID ?

Comment (llorllale) [reply pkg/restapi/rp/operation/operations.go:608]: This should be a protected endpoint. It should be called by someone with the necessary privileges to register new RP tenants.

As for the publicDID - we can have the adapter create it while fulfilling this request.

Comment (llorllale) [reply pkg/restapi/rp/operation/operations.go:608]: @rolsonquadras create #67

**Pull request merged**